src/screens/authentication/Login/index.js

```diff
@@ -20,11 +21,15 @@ const onGoogleButtonPress = async () => {
     const userEmail = signInData.user.email
     if (await AuthenticationService.emailAlreadyHasGoogleSignIn(userEmail)) {
       await AuthenticationService.loginWithGoogleSignInData(signInData)
+      await MessagingService.checkForPermissions()
+      await MessagingService.saveFCMTokenForCurrentUser()
       NavigationService.navigate('MainApp')
     } else {
       const username = await DatabaseService.queryUsernameFromEmail(userEmail)
       if (username) {
         await AuthenticationService.loginWithGoogleSignInData()
+        await MessagingService.checkForPermissions()
+        await MessagingService.saveFCMTokenForCurrentUser()
```

Comment:
Same here, also you're repeating code (calling these two lines in the same order). This should definitely be on its own function. Maybe `enableNotificationsForCurrentUser`.

Reply:
Those functions doesn't throw errors. Also I talked this with Mauro because we believe that login will have a rework for the new wireframes where those functions will only be needed in 1 place...

Reply:
Although it's true that some things will change for the v4 wireframe and the login process can be encapsulated, I second Nahuel here. This piece of information should only have one representation. 

Even if this ends up only needed in one place, then this block of `checkForPermissions` and `saveToken`would still be those two lines. If `saveToken` assumes that push notification were allowed, then it can sustain itself without `checkForPermissions`, but this is not the case as a previous check is always needed. Everytime a `saveToken` is done, it has to be preceded with a `check` (there is some implicit 'coupling' there). Although it's somewhat true that if it only appears in one place, then DRY is not violated (Debatable), but as it stands, Nahuel is right.

Let's decide which part of push notifications is mandatory to have.

-  If the user doesn't allow them so he can't login. If this is the case lets throw an error and check for this at the start of the login process.

- If the user doesn't allow them and he can login. If this is a case a thrown error shouldn't interfere with the login process (log the error or have a nested try/catch take care of that)

- If it doesn't matter if the token is deleted or not, then an error on that function shouldn't block the signing out (log the error or have a nested try/catch take care of that)

- If it does matter that the token is deleted or not, then an error on this function should block the signing out (I doubt this could result in an error unless external factors, but if it happens, then no action is done or a reusable modal for unexpected errors can be shown).

For consistency, can we decide to have all services throw an error and later, in the consumer code, decide if that error does or doesn't have to block the normal flow with nested try/catchs?

@nahudalla @jmbidev

Reply:
Actually, it's not required to call checkForPermissions every time we call saveFCMTokenForCurrentUsers. It is a coincidence that I decided to use those two together. 

checkForPermissions will ask if messaging is enabled and if not will request those permissions.
saveToken will ask if there are permissions to proceed sending the token, but won't request.

I think the DRY was already violated with the redirections to MainApp, and even if we group the two functions in a single function still will be violated since It's in many place while it could be all in the same place. But I think that will be fixed with the new wireframes.

Btw that doesn't mean that those functions can't be on the same function, so I will group them. 

And I agree to throw erros in services for consistency

Reply:
> - If the user doesn't allow them so he can't login. If this is the case lets throw an error and check for this at the start of the login process.
> - If the user doesn't allow them and he can login. If this is a case a thrown error shouldn't interfere with the login process (log the error or have a nested try/catch take care of that)
I think the second option is the correct.


> - If it doesn't matter if the token is deleted or not, then an error on that function shouldn't block the signing out (log the error or have a nested try/catch take care of that)
> - If it does matter that the token is deleted or not, then an error on this function should block the signing out (I doubt this could result in an error unless external factors, but if it happens, then no action is done or a reusable modal for unexpected errors can be shown).

I think this function should allways delete if there is a token set. And prevent user logout if couldn't remove the token. Because if it wasn't removed the device will still receive notifications for that user. So if you are ok with it I'll make the changes needed.

@nahudalla @MPucheta